src/main/java/core/basesyntax/MainApp.java

```diff
@@ -1,5 +1,16 @@
 package core.basesyntax;
 
 public class MainApp {
+    public static void main(String[] args) {
 
+        Machine[] machines = new Machine[3];
+        machines[0] = new Truck();
+        machines[1] = new Bulldozer();
+        machines[2] = new Excavator();
```

Comment:
Consider using abstract references instead of specific ones where possible. This will allow for better polymorphism in your code. For example, use 'Machine truck = new Truck();' instead of 'Truck truck = new Truck();'.